src/Domain/Python/ElementLogicalCoordinates.cpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Python/ElementLogicalCoordinates.hpp"
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/IdPair.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/BlockLogicalCoordinates.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementLogicalCoordinates.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/Structure/BlockId.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace py = pybind11;
+
+namespace domain::py_bindings {
+
+namespace {
+template <size_t Dim>
+void bind_element_logical_coordinates_impl(py::module& m) {  // NOLINT
+  m.def(
+      "element_logical_coordinates",
+      [](const std::vector<ElementId<Dim>>& element_ids,
+         const std::vector<std::optional<
+             IdPair<domain::BlockId,
+                    tnsr::I<double, Dim, typename Frame::BlockLogical>>>>&
+             block_coord_holders) {
+        const auto element_logical_coords =
+            element_logical_coordinates(element_ids, block_coord_holders);
+        // Transform result to Python types
```

Comment:
is what's transformed here tensor->array? If not, it would be helpful to state what is actually transformed and why. It's not clear to me

Reply:
With the Tensor bindings I removed this code and bound the return value struct instead